app/controllers/user_sessions_controller.rb

```diff
@@ -26,8 +26,14 @@ def handle_social_login_flow(auth)
     unless params[:hash_params].to_s.empty?
       hash_params = URI.parse("#" + params[:hash_params]).to_s
     end
-
-    if signed_in?
+    @user = User.where(email: auth["info"]["email"]) ? User.find_by(email: auth["info"]["email"]) : User.find(@identity.user.id)
+    if @user&.status&.zero?
+      flash[:error] = I18n.t('user_sessions_controller.user_has_been_banned', username: @user.username).html_safe
+      redirect_to root_url
```

Comment:
redirection is not handled here . Kindly handle it

Reply:
What do you mean?

Reply:
Yeah I think I fixed it